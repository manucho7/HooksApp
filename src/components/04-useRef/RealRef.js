import React, { useState } from 'react'
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

import '../02-useEffect/effects.css';

export const RealRef = () => {

    const [show, setShow] = useState(false);

    return (
        <div>
            <h1>RealRef example</h1>
            <hr />

            { show && <MultipleCustomHooks /> }

            <button
                className="btn btn-primary mt-5"
                onClick= { () => {
                    setShow( !show );
                }}
            >
                Show/Hide
            </button>

        </div>
    )
}
