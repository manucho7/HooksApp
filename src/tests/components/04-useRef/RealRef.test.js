import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import '@testing-library/dom';
import { RealRef } from '../../../components/04-useRef/RealRef';


describe('Pruebas en <RealRef />', () => {

    const wrapper = shallow( <RealRef /> );
    
    test('debe mostrarse correctamente ', () => {
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'MultipleCustomHooks' ).exists() ).toBe(false);
        console.log( wrapper.html());
    });
    
    test('debe de mostrar el componente <MultipleCustomHooks />', () => {
        wrapper.find('button').simulate('click');
        expect( wrapper.find( 'MultipleCustomHooks' ).exists() ).toBe(true);
    });
})
