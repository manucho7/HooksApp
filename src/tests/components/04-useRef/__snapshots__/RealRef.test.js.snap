// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealRef /> debe mostrarse correctamente  1`] = `
<div>
  <h1>
    RealRef example
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
